extensions/amp-document-recommendations/validator-amp-document-recommendations.protoascii

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+
+tags: {  # amp-document-recommendations
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-document-recommendations"
+    allowed_versions: "0.1"
+    allowed_versions: "latest"
+  }
+}
+tags: {  # amp-document-recommendations: json config
+  html_format: AMP
+  tag_name: "SCRIPT"
+  spec_name: "amp-document-recommendations extension .json configuration"
+  requires_extension: "amp-document-recommendations"
+  mandatory_parent: "AMP-DOCUMENT-RECOMMENDATIONS"
+  attrs: {
+    name: "recommendations"
+    mandatory: true
+    value_casei: "application/json"
+    dispatch_key: NAME_VALUE_PARENT_DISPATCH
+  }
+  spec_url: "https://www.ampproject.org/docs/reference/components/amp-document-recommendations"
+}
+tags: {  # <amp-document-recommendations>
+  html_format: AMP
+  tag_name: "AMP-DOCUMENT-RECOMMENDATIONS"
+  mandatory_parent: "BODY"
```

Comment:
Is there a reason this needs to have parent of `body` and why it must be the last child of `body`?

Reply:
This is a restriction that we might want to remove or relax it in the future, but forcing those two conditions reduces potentially unexpected behaviors (e.g. no footer on the first article as there are others in between, but the following article having a footer in the middle as it does not load other articles recursively).

Adding support to hide things like footer in all but the last article and headers in all but the first can probably led to the relaxation of this in followup changes.

Any particular concerns with this restrictions?

Reply:
Yes, it means no other component can also be last in the body. It makes this component have some significance among all components. @aghassemi thoughts?

Reply:
Spoke with @aghassemi to clarify intent. Thank you.